src/combinators/mod.rs

```diff
@@ -1022,27 +982,34 @@ where Gen: Fn(W) -> GenResult<(W, Tmp)>,
 //text lowerhex
 
 impl Skip for &mut [u8] {
-    fn skip(self, len: usize) -> Result<Self, GenError> {
-        if self.len() < len {
-            Err(GenError::BufferTooSmall(len - self.len()))
+    fn skip(mut s: WriteContext<Self>, len: usize) -> Result<WriteContext<Self>, GenError> {
+        if s.get_ref().len() < len {
+            Err(GenError::BufferTooSmall(len - s.get_ref().len()))
         } else {
-            Ok(&mut self[len..])
+            use std::mem;
+            let slice = mem::replace(s.get_mut(), &mut []);
+            *s.get_mut() = &mut slice[len..];
+            s.set_position(s.position() + len as u64);
+            Ok(s)
```

Comment:
Why the complexity? Isn't this enough?

```
s.write = s.write[len..];
s.position += len as u64;
Ok(s)
```

Reply:
I thought so too but `rustc` does not like the first line of that.

Reply:
Maybe `Ok(WriteContext::new(s.write[len..], s.position + len as u64))` would make it happy as all of it is consumed in one line?

Reply:
Good idea, that should work. The problem was `s.write = s.write[len..]` itself btw, `std` also has the workaround I used above in e.g. the `ChunksMut` iterator on slices.